### PR TITLE
AI Nerf(Adds signal jamming) + Codebase QoL

### PR DESCRIPTION
**Pull request by MLGTASTICa**
## About The Pull Request
Adds signal jamming component along with SS to coordiante it
Gives the AI a custom click handler that won't let it perform acts when a zone is jammed
Antagonists can buy a portable or stationary version of the jammer for 3 , respectively 15 TC
AI has an field of 16 tiles around its core in which it cannot be jammed
The range of the portable jammer is 8 tiles,  the  stationary one has 24 tiles of range
The lifetime of the portable jammer is 666 seconds with a 100S cell,.
The lifetime of the stationary jammer is 1000 seconds with a 1000L cell.
Added a overlay manager component for general handling overlays
Added a new signal , COMSIG_ATOM_CONTAINERAIZED,  sent when an atom is put inside of another, for registering onto the highest and most relevant atom that will receive all signals
Added a new signal , COMSIG_MOVABLE_ZCHANGED , sent when an atom is moved to another Z-level.
Fixed a bug with dragging  self into an AI , and the hud not being rebuilt.

## Why It's Good For The Game
Gives antagonists more counterplay against AI without needing to actually always kill the AI
codebase QOL  + new components always good
## Testing
Tested the reworked quick equip , jamming purchasing , AI being able to click , item transfers between clothing bags and backpacks (doesn't work and adding functionality for this is not within the scope)
https://discord.com/channels/255035529085583360/265411250341543936/1093917214211588227

## Changelog
:cl:
add: Added 2 new traitor gadgets for  preventing AI control for an area , the portabla jammer , for 3 TC,  which stops any AI click in a 8 tile radius, with a lifetime of 666 seconds for a 100S cell , and the big portable jammer , 15 TC, stops in a 24 tile radius , with a lifetime of 1000 S for a 1000L cell.
code: added an Overlay manager component
code: Added a new comsig signal , COMSIG_ATOM_CONTAINERIZED , for when an item is put inside any storage object, has as the argument the top-most container
code: Added a new comsig signal , COMSIG_MOVABLE_ZCHANGED,  for when an movable atom is sent to another z-level.
qol: Item dragging now works towards any slot , not just hands , for clothing attire.
fix: Fixed AI's not receiving a hud  when a ghost was dragged inside
/:cl:

### CODE REVIEW

**Comment by TheShown911** [comment]
Linters fail

File: code/controllers/subsystems/jamming.dm [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	var/list/datum/component/jamming/active_jammers
+
+/datum/controller/subsystem/jamming/Initialize(start_timeofday)
+	active_jammers = new /list(world.maxz)
+	for(var/i = 1, i < world.maxz; i++)
+		active_jammers[i] = list()
+	. = ..()
+
+/datum/controller/subsystem/jamming/proc/IsPositionJammed(turf/location, signalStrength)
+	for(var/datum/component/jamming/jammer in active_jammers[location.z])
```
Comment (Pink-Chink) [inline code/controllers/subsystems/jamming.dm:17]: `active_jammers` is declared as a list of jammer datums, but it looks like this is expecting it to be nested lists.

File: code/game/machinery/jammers.dm [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+		to_chat(user , SPAN_NOTICE("The charge in [src]'s cell is too low to start jamming"))
+	overlay_manager.removeOverlay(OVERKEY_JAMMER_OFF)
+	overlay_manager.addOverlay(OVERKEY_JAMMER_ON, mutable_appearance(icon, "jammermobover_on"))
+	our_jammer.Toggle()
+	to_chat(user,  "You toggle the [src] on")
+	cell.charge -= power_usage
+	START_PROCESSING(SSobj, src)
+
+#undef OVERKEY_JAMMER_OFF
+#undef OVERKEY_JAMMER_OFF
```
Comment (Pink-Chink) [inline code/game/machinery/jammers.dm:137]:
```suggestion
#undef OVERKEY_JAMMER_OFF
```

File: code/game/objects/items.dm [hunk 3]
```diff
@@ -276,12 +276,13 @@
 
 // called when this item is removed from a storage item, which is passed on as S. The loc variable is already set to the new destination before this is called.
 /obj/item/proc/on_exit_storage(obj/item/storage/the_storage)
-	SEND_SIGNAL_OLD(the_storage, COMSIG_STORAGE_TAKEN, src, the_storage)
+	SEND_SIGNAL(the_storage, COMSIG_STORAGE_TAKEN, src, the_storage)
```
Comment (Pink-Chink) [inline code/game/objects/items.dm:279]:
```suggestion
	SEND_SIGNAL(the_storage, COMSIG_STORAGE_TAKEN, src)
```
The first arg of the new `SEND_SIGNAL()` is passed as the first arg in the registered proc. Same thing applies to `on_enter_storage()`.

**Comment by TheShown911** [comment]
Linters still fail

**Review by hyperioo — COMMENTED** [review]
Seems ok to deploy on Monday for (live?) testing

**Review by TheShown911 — APPROVED** [review]
PR to add AI signals jamming adds a new SS, overlay manager component, reworks containerization of signals and AI click handlers. Maybe a bit much isn't it, but you know what fine. Nothing is bad on it's own just please atomize everything next time, if just one of these breaks the whole PR will need to be reverted.

**Pull request merged**